src/ngd.ts

```diff
@@ -0,0 +1,197 @@
+// src/handlers/ngd.ts
+
+import { request } from "./utils/ngd/request.js";
+import { get } from "./utils/ngd/request.js";
+import { Feature, type FeatureCollection } from "geojson";
+import { buildUrl } from "./utils/ngd/url.js";
+import { validateParams } from "./utils/validate.js";
+import { BBox, Config } from "./types.js";
+import { initialiseConfig } from "./utils/config.js";
+import fetch from "node-fetch";
+import { getCRS } from "./utils/crs.js";
+
+export { ngd };
+
+// Types
+interface NGDExtent {
+  spatial: {
+    bbox: number[][];
+    crs: string;
+  };
+  temporal: {
+    interval: string[][];
+    trs: string;
+  };
+}
+
+interface NGDLink {
+  href: string;
+  rel: string;
+  type: string;
+  title: string;
+}
+
+interface NGDCollection {
+  id: string;
+  title: string;
+  description: string;
+  crs: string[];
+  storageCrs: string;
+  itemType: string;
+  extent: NGDExtent;
+  links: NGDLink[];
+}
+
+interface NGDCollections {
+  links: NGDLink[];
+  collections: NGDCollection[];
+}
+
+interface NGDQueryables {
+  $schema: string;
+  $id: string;
+  type: string;
+  title: string;
+  description: string;
+  properties: any;
+}
+
+interface NGDSchema {
+  $schema: string;
+  $id: string;
+  type: string;
+  title: string;
+  description: string;
+  extent: NGDExtent;
+  properties: any;
+}
+
+async function requestNGD(config: Config): Promise<FeatureCollection> {
+  return await request(config);
+}
+
+const ngd = {
+  /**
+   * Get NGD features.
+   *
+   * @param {string} apiKey - A valid OS Data Hub key
+   * @param {string} collectionId - A known collection ID
```

Comment:
Point people towards where they can find the collection ids

Reply:
Added